console/console-init/ui/src/modules/project-detail/components/ProjectCertificateList/CertificateForm.tsx

```diff
@@ -0,0 +1,217 @@
+/*
+ * Copyright 2020, EnMasse authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+
+import React, { useState, useEffect } from "react";
+import {
+  Card,
+  CardBody,
+  Form,
+  FormGroup,
+  TextInput,
+  Grid,
+  GridItem,
+  Split,
+  SplitItem,
+  ActionGroup,
+  Button,
+  Switch,
+  DropdownPosition
+} from "@patternfly/react-core";
+import { DropdownWithToggle } from "components";
+import { css, StyleSheet } from "@patternfly/react-styles";
+import { getCertificateLabelByKey } from "modules/project-detail/utils/utils";
+import { algorithmTypeOptions } from "modules/project-detail/utils/constant";
+import { IProjectCertificate } from "./ProjectCertificateList";
+
+const styles = StyleSheet.create({
+  dropdown_align: { display: "flex" },
+  dropdown_toggle_align: { flex: "1" }
+});
+
+export interface ICertificateFormProps {
+  certificate?: IProjectCertificate;
+  setOnEditMode: React.Dispatch<React.SetStateAction<boolean>>;
+  id: string;
+}
+
+export const CertificateForm: React.FunctionComponent<ICertificateFormProps> = ({
+  setOnEditMode,
+  certificate,
+  id
+}) => {
+  const [subjectDn, setSubjectDn] = useState<string>();
```

Comment:
can you try to store all the values in single state as a object.

Reply:
Done.